parcels/particlefile/baseparticlefile.py

```diff
@@ -155,6 +155,9 @@ def close_netcdf_file(self):
         self.dataset.close()
 
     def _create_trajectory_file(self, fname, data_shape):
+        # ==== This is the line that currently fails the tests - the new netcdf package seems to
+        # ==== use a different signature, making that call fail with an internal pedantic-warning
+        # ==== interpretation of the underlying HDF5 library.
         self.dataset = netCDF4.Dataset(fname, "w", format="NETCDF4")
```

Comment:
@erikvansebille this is where the error actually originates and I assume it;s linked to the `format` string there.

Reply:
seems that the current conda or pip-distributed packages has some bad internal linking, as already previously experienced with recent versions: https://stackoverflow.com/questions/57842565/hdf5-library-version-error-hdf5-ver-1-10-4

Our issue is similar: the `netcdf-python` package (at least the recent version) is built against `hdflib v. 1.10.7`, whereas the library dependency distributed by conda-forge is `hdflib v. 1.10.6`. I'll try to track that error back by comparing recent successful pytest logs here on github-parcels with this error log.

Reply:
as suspected, the head versions of netcdf must have changed recently. here is the comparison from the last successful merge and the current log:

last merge
![hdf5_working](https://user-images.githubusercontent.com/6020531/120361060-ba245f80-c309-11eb-827a-4d443ae53b59.png)

this issue
![hdf5_not_working](https://user-images.githubusercontent.com/6020531/120361093-c4465e00-c309-11eb-8dc6-2ecc99e8bef4.png)

I'll try to see if I can fix the hdf version in the `environment_py_osg.yml` and `environment_py3p6_linux.yml` files.

Reply:
And what if we change the `format` string to something else than `"NETCDF4"`?

Reply:
Okay, so: the fix worked. 
@erikvansebille : von't be any good because one of the base libraries the netcdf4 package is built on is broken in its C++ dependencies. In other words: the issue is not just with the `format` string (as I originally assumed), it's with the entire function call. It will be fixed as soon as the people maintaining `netcdf`fix their conda-forge git with their C++ dependency. They basically just need to rerun the conda-forge with the new HDF5 version ... that's why the fix is just 'temporary'.

Reply:
[do not resolve now]
by the time the PR is done, I'll revert the hdf5 version constraint again to see if the netcdf-package on conda-forge is fixed and if we so then can drop that quick-hack.